src/main/java/duke/helpers/TaskList.java

```diff
@@ -0,0 +1,135 @@
+package duke.helpers;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * TaskList class: Structure that stores each task.
+ */
+public class TaskList {
+    private List<Task> tasks;
+
+    /**
+     * TaskList constructor: Structure that stores each task.
+     */
+    public TaskList() {
+        tasks = new ArrayList<Task>();
+    }
+
+    public Task getTask(int index) {
+        return this.tasks.get(index);
+    }
+
+    public int getSize() {
+        return tasks.size();
+    }
+
+    /**
+     * Adds task into the TaskList structure.
+     */
+    public void addTask(Task task) {
+        this.tasks.add(task);
+    }
+
+    /**
+     * Reads in a char (length 1 string), and outputs whether it is checked or not.
+     *
+     * @param mark Reads char, could be X or white space.
+     */
+    public boolean checkMark(String mark) {
+        if (mark.equals("X")) {
+            return true;
+        } else return false;
+    }
+
+    /**
+     * Reads in a index, which is the index to be removed in the structure.
+     *
+     * @param index Indicates which index of the structure to delete.
+     */
+    public void removeTask(int index) {
+        this.tasks.remove(index);
+    }
+
+    /**
+     * Adds a task object from the load storage into this TaskList structure.
+     *
+     * @param line Feeds in the line from load storage, interpretes it into a task.
+     */
+    public void addLine(String line) {
+        if (line.substring(1,2).equals("T")) {
+            String todo = line.substring(8);
+            Todo newTodo = new Todo(todo, this.checkMark(line.substring(5,6)));
+            this.addTask(newTodo);
+        }
+        if (line.substring(1,2).equals("D")) {
+            String deadline = line.substring(8);
+            Deadline newDead = new Deadline(deadline, this.checkMark(line.substring(5,6)));
+            this.addTask(newDead);
+        }
+        if (line.substring(1,2).equals("E")) {
```

Comment:
I feel there are some unnecessary repetitions in this method. I like to structure it like this: 
`String str = line.substring(8);` and `Task newTask;`  declared before the if statement. A value to newTask (like `newTask = new Todo(todo, this.checkMark(line.substring(5,6)));` in the if else statement. Then at the end `addTask(task);` after the if statement closes.